internal/hmyapi/apiv1/blockchain.go

```diff
@@ -765,3 +768,20 @@ func (s *PublicBlockChainAPI) GetSuperCommittees() (*quorum.Transition, error) {
 	}
 	return nil, errNotBeaconChainShard
 }
+
+// GetTotalSupply ..
+func (s *PublicBlockChainAPI) GetTotalSupply() (numeric.Dec, error) {
+	if s.b.GetShardID() == shard.BeaconChainShardID {
+		return numeric.NewDec(initSupply), nil
+	}
+	return numeric.NewDec(0), errNotBeaconChainShard
```

Comment:
seems this error is to be used for any RPC that are to be used for requests made to beacon chain nodes only.
can you give me some context on what these rpc endpoints are to be only served by nodes that are in shard 0?

Reply:
Hm, actually they can be used from any shard. Sorry. Some staking RPCs can be only used from beaconchain.